docs/src/adr/015adr-trace.md

```diff
@@ -0,0 +1,403 @@
+Missionaries == { "m1_OF_PERSON", "m2_OF_PERSON" }
+Cannibals == { "c1_OF_PERSON", "c2_OF_PERSON" }
+
+VARIABLES
+    \* @type: Str;
+    bank_of_boat,
+    \* @type: Str -> Set(PERSON);
+    who_is_on_bank
+
+INSTANCE MissionariesAndCannibalsTyped
+
+NoSolution ==
+    who_is_on_bank["E"] /= {}
+=============================================================================
+```
+
+By checking the invariant `NoSolution`, we obtain the following counterexample
+in TLA+:
+
+```tla
+---------------------------- MODULE counterexample ----------------------------
+
+EXTENDS MC_MissionariesAndCannibalsTyped
+
+(* Constant initialization state *)
+ConstInit == TRUE
+
+(* Initial state *)
+State0 ==
+  bank_of_boat = "E"
+    /\ who_is_on_bank
+      = "E"
+          :> { "c1_OF_PERSON", "c2_OF_PERSON", "m1_OF_PERSON", "m2_OF_PERSON" }
+        @@ "W" :> {}
+
+(* Transition 0 to State1 *)
+State1 ==
+  bank_of_boat = "W"
+    /\ who_is_on_bank
+      = "E" :> { "c1_OF_PERSON", "m1_OF_PERSON" }
+        @@ "W" :> { "c2_OF_PERSON", "m2_OF_PERSON" }
+
+(* Transition 0 to State2 *)
+State2 ==
+  bank_of_boat = "E"
+    /\ who_is_on_bank
+      = "E" :> { "c1_OF_PERSON", "m1_OF_PERSON", "m2_OF_PERSON" }
+        @@ "W" :> {"c2_OF_PERSON"}
+
+(* Transition 0 to State3 *)
+State3 ==
+  bank_of_boat = "W"
+    /\ who_is_on_bank
+      = "E" :> {"c1_OF_PERSON"}
+        @@ "W" :> { "c2_OF_PERSON", "m1_OF_PERSON", "m2_OF_PERSON" }
+
+(* Transition 0 to State4 *)
+State4 ==
+  bank_of_boat = "E"
+    /\ who_is_on_bank
+      = "E" :> { "c1_OF_PERSON", "c2_OF_PERSON" }
+        @@ "W" :> { "m1_OF_PERSON", "m2_OF_PERSON" }
+
+(* Transition 0 to State5 *)
+State5 ==
+  bank_of_boat = "W"
+    /\ who_is_on_bank
+      = "E" :> {}
+        @@ "W"
+          :> { "c1_OF_PERSON", "c2_OF_PERSON", "m1_OF_PERSON", "m2_OF_PERSON" }
+
+(* The following formula holds true in the last state and violates the invariant *)
+InvariantViolation == who_is_on_bank["E"] = {}
+
+================================================================================
+(* Created by Apalache on Wed Dec 22 09:18:50 CET 2021 *)
+(* https://github.com/informalsystems/apalache *)
+```
+
+The above counterexample looks very simple and natural, if the reader knows
+TLA+. In our experience, these examples look alien to engineers, who are not
+familiar with TLA+. It is unfortunate, since the counterexamples have a very
+simple shape:
+
+ 1. They are simply sequences of states.
+
+ 1. Every state is a mapping from variable names to expressions that do not
+ refer to other variables.
+
+ 1. The expressions are using a very small subset of TLA operators:
+
+   1. Integer and string literals.
+
+   1. Set constructor, sequence/tuple constructor, record constructor.
+
+   1. TLC operators over functions: `:>` and `@@`.
+
+In hindsight, the above expressions are not very far from the JSON format.
+As many engineers know [JSON][], it seems natural to write these counterexamples
+in JSON.
+
+## Options
```

Comment:
Seems like the options here don't include the option you're proposing? I think that's valuable because it lets us clearly compare the pros and cons of the options not selected against the option which is.

Reply:
Fixed.